Kotlin floor() method implementation (#3555)
Codecademy/docs: content/kotlin/concepts/math-methods/terms/floor/floor.md

---
Title: 'floor()'

> Description: 'The `floor()` method in Kotlin is used to round a floating-point to a nearest lowest integer value'

Reviewer: ```suggestion
Description: 'Returns the nearest integer that is lower than or equal to the value passed.'
```

Subjects:
  - 'Computer Science'
Tags:
  - 'Arithmetic'
  - 'Methods'
  - 'Floor'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The **`floor()`** function returns a rounded number down to the nearest integer.

Reviewer: ```suggestion
The **`floor()`** function takes a value and returns the nearest integer that is lower than or equal to the value given.
```

The result returns a value type of `float`.

## Syntax

```pseudo

> Math.floor(x:Double)

Reviewer: ```suggestion
import kotlin.math.*
floor(x)
```

Reviewer: I believe `Math` calls the java library

```

> Where `x` is the number to be rounded down to the nearest integer.

Reviewer: ```suggestion
Where `x` is the number of type double to be rounded down to the nearest integer.
```

## Example

```kotlin

> println(Math.floor(3.4))

Reviewer: ```suggestion
import kotlin.math.*
println(floor(3.4))
```

// Output: 3
```

The result will be a float value of `3` as the function rounds `3.4` down to `3`.

> ## Codebyte Example

Reviewer: kotlin is not currently supported by the Codecademy compiler so I would suggest removing this example
